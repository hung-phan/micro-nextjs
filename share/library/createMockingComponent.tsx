import * as _ from "lodash";
import * as React from "react";

export default (mockTagName: string, props: any[] = []) =>
  (componentProps: { [key: string]: any }) =>
    (
      <div>{`${mockTagName} ${JSON.stringify(
        _.pick(componentProps, props)
      )}`}</div>
    );
