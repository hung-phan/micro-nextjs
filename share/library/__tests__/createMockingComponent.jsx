import React from "react";
import { render } from "enzyme";
import createMockingComponent from "../createMockingComponent";

test("createMockingComponent", () => {
  const Component = createMockingComponent("MockingComponent", ["prop1"]);

  expect(render(<Component prop1="value1" prop2="value2" />)).toMatchSnapshot();
});
