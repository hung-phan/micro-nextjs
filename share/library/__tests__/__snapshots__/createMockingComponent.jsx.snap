// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMockingComponent 1`] = `
<div>
  MockingComponent {"prop1":"value1"}
</div>
`;
